backend/Terraform/modules/management_api/rest_api_getway.tf

```diff
@@ -0,0 +1,158 @@
+resource "aws_api_gateway_rest_api" "management_api" {
+  name        = var.api_name
+  endpoint_configuration {
+    types = ["REGIONAL"]
+  }
+}
+
+resource "aws_api_gateway_resource" "announcement_lambda" {
+  rest_api_id = aws_api_gateway_rest_api.management_api.id
+  parent_id   = aws_api_gateway_rest_api.management_api.root_resource_id
+  path_part   = "announcement_lambda"
+}
+
+resource "aws_api_gateway_method" "proxy" {
+  rest_api_id = aws_api_gateway_rest_api.management_api.id
+  resource_id = aws_api_gateway_resource.announcement_lambda.id
+  http_method = "POST"
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_integration" "lambda_integration" {
+  rest_api_id = aws_api_gateway_rest_api.management_api.id
+  resource_id = aws_api_gateway_resource.announcement_lambda.id
+  http_method = aws_api_gateway_method.proxy.http_method
+  integration_http_method = "POST"
+  type = "AWS"
+  uri = aws_lambda_function.name_of_the_lambda.invoke_arn #change to the real name of the lambda
+}
+
+resource "aws_api_gateway_method_response" "proxy" {
+  rest_api_id = aws_api_gateway_rest_api.management_api.id
+  resource_id = aws_api_gateway_resource.announcement_lambda.id
+  http_method = aws_api_gateway_method.proxy.http_method
+  status_code = "200"
+
+    //cors section
+  response_parameters = {
+    "method.response.header.Access-Control-Allow-Headers" = true,
+    "method.response.header.Access-Control-Allow-Methods" = true,
+    "method.response.header.Access-Control-Allow-Origin" = true
+  }
+
+}
+
+resource "aws_api_gateway_integration_response" "proxy" {
+  rest_api_id = aws_api_gateway_rest_api.management_api.id
+  resource_id = aws_api_gateway_resource.announcement_lambda.id
+  http_method = aws_api_gateway_method.proxy.http_method
+  status_code = aws_api_gateway_method_response.proxy.status_code
+
+    //cors
+  response_parameters = {
+    "method.response.header.Access-Control-Allow-Headers" =  "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,Content-Length,DateX-Amz-Apigw-Id,X-Amzn-Errortype,X-Amzn-Requestid '",
+    "method.response.header.Access-Control-Allow-Methods" = "'GET,OPTIONS,POST,PUT'",
+    "method.response.header.Access-Control-Allow-Origin" = "'*'"
+}
+  
+  depends_on = [
+    aws_api_gateway_method.proxy,
```

Comment:
You can remove `aws_api_gateway_method.proxy` from the `depands on` block as it is referenced in line 48